src/app/pages/Order.jsx

```diff
@@ -0,0 +1,390 @@
+            onChange={(e) => {
+              setSelectedAddress(e.target.value);
+            }}
+            label="Endereço"
+          >
+            {addresses.length ? (
+              addresses.map((addr) => (
+                <MenuItem key={addr.id} value={addr}>
+                  {addr.address}
+                </MenuItem>
+              ))
+            ) : (
+              <MenuItem value={selectedAddress}>
+                <em>Não há opções</em>
+              </MenuItem>
+            )}
+          </Select>
+        </FormControl>
+        <Button disabled={disabled} onClick={onNewAddress} variant="outlined">
+          Novo Endereço
+        </Button>
+
+        {!!selectedAddress.length && (
+          <Typography>{`${selectedAddress?.address}, ${selectedAddress.number}, ${selectedAddress.cep} - ${selectedAddress.city}, ${selectedAddress.state}`}</Typography>
+        )}
+      </Grid>
+    </Grid>
+  );
+}
+
+function DeliverySelect({ cep, onSelect }) {
+  const [shippings, setShippings] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  const clearSelection = () => onSelect(0);
+
+  useEffect(() => {
+    clearSelection();
+    const getShippingMethods = async () => {
+      if (cep) {
+        setLoading(true);
+        const response = await client.shipping.listForCep(cep);
+        setShippings(response);
+        setLoading(false);
+      }
+    };
+    getShippingMethods();
+  }, [cep]);
+
+  return (
+    <Grid>
+      {loading ? (
+        <CircularProgress />
+      ) : (
+        !!shippings.length && (
+          <Grid>
+            <Typography variant="h4">Valor Frete</Typography>
+            <RadioGroup
+              defaultValue="female"
+              onChange={(e) => {
+                const selectedShipping = shippings.find(
+                  (shipng) => shipng.Codigo === e.target.value,
+                );
+                onSelect(parseFloat(selectedShipping.Valor));
+              }}
+            >
+              {shippings.map((shipping, index) => (
+                <DeliveryRadioButton
+                  key={shipping.Codigo}
+                  value={shipping.Codigo}
+                  shippingMethod={shipping}
+                  number={index}
+                />
+              ))}
+            </RadioGroup>
+          </Grid>
+        )
+      )}
+    </Grid>
+  );
+}
+
+function DeliveryRadioButton({ shippingMethod, value, number }) {
+  return (
+    <FormControlLabel
+      value={value}
+      control={<Radio />}
+      label={
+        <>
+          <Typography display="inline">
+            {`[Correios] Opção ${number + 1}  - Entrega em até ${
+              shippingMethod.PrazoEntrega
+            } dias úteis `}
+          </Typography>
+          <Typography display="inline" color="success.main">
+            {formatCurrency(parseFloat(shippingMethod.Valor))}
+          </Typography>
+        </>
+      }
+    />
+  );
+}
+
+function OrderItem({ item }) {
```

Comment:
Coloca ou na foto ou no nome, ou nos dois para quando clicar redirecionar para a pagina do produto, para mi me muito util, para antes de comprar, confirmar se o que eu to comprando esta correto